### PR TITLE
Blog post: Rapid application development with Gradle and Open Liberty's dev mode

### PR DESCRIPTION
**Pull request by rzgry**
https://github.com/OpenLiberty/blogs/issues/254

### CODE REVIEW

File: drafts/liberty-dev-mode-gradle.adoc [hunk 1]
```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: "Open Liberty development mode with Gradle"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/u/28316667
+author_github: https://github.com/rzgry
+seo-title: Open Liberty development mode with Gradle - OpenLiberty.io
+seo-description: Develop Open Liberty applications quicker using Open Liberty development mode with Gradle.
+blog_description: "Develop Open Liberty applications quicker using Open Liberty development mode with Gradle."
```
Comment (ericglau) [inline drafts/liberty-dev-mode-gradle.adoc]: "Open Liberty" is used too many times, perhaps change to `Develop Java applications quicker using Open Liberty development mode with Gradle.` for both SEO and blog description.

File: drafts/liberty-dev-mode-gradle.adoc [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+---
+= Open Liberty development mode for Gradle
+Ryan Zegray <https://github.com/rzgry>
+
+Open Liberty development mode, or dev mode, now works with Gradle projects. Open Liberty dev mode allows you to develop applications with any text editor or IDE by providing hot reload and deployment, on demand testing, and debugger support. Your code is automatically compiled and deployed to your running server, making it easy to iterate on your changes. You can run tests on demand or even automatically so that you can get immediate feedback on your changes. You can also attach a debugger at any time to debug your running application.
+
+If you would like to give dev mode with Gradle a try you can follow the steps in our https://github.com/OpenLiberty/demo-devmode[demo-devmode] sample project.
```
Comment (ericglau) [inline drafts/liberty-dev-mode-gradle.adoc]: "dev mode with Gradle a try" sounds awkward, sentence needs a comma, and it should reference "the" rather than "our" sample project.
Perhaps:
`If you would like to try out dev mode with Gradle, you can follow the steps in the https://github.com/OpenLiberty/demo-devmode[demo-devmode] sample project.`

File: drafts/liberty-dev-mode-gradle.adoc [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+You can quit dev mode at any time by pressing CTRL+C in the terminal, or type `q` and press Enter.
+
+== VS Code extension
+
+The https://marketplace.visualstudio.com/items?itemName=Open-Liberty.liberty-dev-vscode-ext[Open Liberty Tools VS Code extension] has also been updated to support Gradle projects. This optional extension lets you interact with dev mode entirely within VS Code. You can start Open Liberty in development mode, make changes to your application while the server is up, run tests and view results, and even debug the application without leaving the editor.
+
+image::/img/blog/liberty-dev-vscode-gradle.png[Liberty Dev Dashboard, align="left", width=400]
+
+For more information on the Open Liberty Tools VS Code extension you can read about it in its link:/blog/2019/11/13/liberty-dev-mode-vscode.html[own blog post].
```
Comment (ericglau) [inline drafts/liberty-dev-mode-gradle.adoc]: Needs a comma.  Change link text to be more descriptive on what the link is.
e.g. `For more information on the Open Liberty Tools VS Code extension, you can read about it in the link:/blog/2019/11/13/liberty-dev-mode-vscode.html[Open Liberty dev mode with VS Code blog post].`

**Comment by ericglau** [comment]
Update the screenshot using the revised ordering of the pop up menu items as per https://github.com/OpenLiberty/liberty-dev-vscode-ext/pull/46, which is available in the latest version of the VS Code extension

**Comment by ericglau** [comment]
Looks good to me.  @lauracowen please proceed with review.

File: drafts/liberty-dev-mode-gradle.adoc [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+seo-title: Open Liberty development mode with Gradle - OpenLiberty.io
+seo-description: Develop Java applications quicker using Open Liberty development mode with Gradle.
+blog_description: "Develop Java applications quicker using Open Liberty development mode with Gradle."
+---
+= Open Liberty development mode for Gradle
+Ryan Zegray <https://github.com/rzgry>
+
+Open Liberty development mode, or dev mode, now works with Gradle projects. Open Liberty dev mode allows you to develop applications with any text editor or IDE by providing hot reload and deployment, on demand testing, and debugger support. Your code is automatically compiled and deployed to your running server, making it easy to iterate on your changes. You can run tests on demand or even automatically so that you can get immediate feedback on your changes. You can also attach a debugger at any time to debug your running application.
+
+If you would like to try out dev mode with Gradle, you can follow the steps in the https://github.com/OpenLiberty/demo-devmode[demo-devmode] sample project.
```
Comment (lauracowen) [inline drafts/liberty-dev-mode-gradle.adoc]: I'm not sure what the reader is expected to do here. If they follow the steps in the sample project, are they supposed to ignore the rest of this blog post? What does the blog post provide that the sample project doesn't, and vice versa?

File: drafts/liberty-dev-mode-gradle.adoc [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+author_picture: https://avatars3.githubusercontent.com/u/28316667
+author_github: https://github.com/rzgry
+seo-title: Open Liberty development mode with Gradle - OpenLiberty.io
+seo-description: Develop Java applications quicker using Open Liberty development mode with Gradle.
+blog_description: "Develop Java applications quicker using Open Liberty development mode with Gradle."
+---
+= Open Liberty development mode for Gradle
+Ryan Zegray <https://github.com/rzgry>
+
+Open Liberty development mode, or dev mode, now works with Gradle projects. Open Liberty dev mode allows you to develop applications with any text editor or IDE by providing hot reload and deployment, on demand testing, and debugger support. Your code is automatically compiled and deployed to your running server, making it easy to iterate on your changes. You can run tests on demand or even automatically so that you can get immediate feedback on your changes. You can also attach a debugger at any time to debug your running application.
```
Comment (lauracowen) [inline drafts/liberty-dev-mode-gradle.adoc]: Nice intro.

File: drafts/liberty-dev-mode-gradle.adoc [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+== Automatically recompiling and deploying to the running server
+
+Once dev mode is started, source code changes are automatically compiled and deployed to the running server. If any features are added to the `server.xml` file, they are installed automatically.
+
+For example, adding the the `mpHealth-2.1` feature to server.xml to enable health checks will make the health endpoint accessible at http://localhost:9080/health.
+
+
+image::/img/blog/dev_mode_server_xml.png[mp-health Server XML]
+image::/img/blog/dev_mode_health_check_1.png[MicroProfile Health check]
```
Comment (ericglau) [inline drafts/liberty-dev-mode-gradle.adoc]: Resize these images since they are too large.  You could resize by adding something like `, width=400]` at the end of the image tag.

File: drafts/liberty-dev-mode-gradle.adoc [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+            SystemResource.class.getSimpleName() + " liveness check")
+                                  .withData("memory used", memUsed)
+                                  .withData("memory max", memMax)
+                                  .state(memUsed < memMax * 0.9).build();
+    }
+    
+}
+----
+
+image::/img/blog/dev_mode_health_check_2.png[MicroProfile Health check]
```
Comment (ericglau) [inline drafts/liberty-dev-mode-gradle.adoc]: Resize this image as per the other comment.

File: drafts/liberty-dev-mode-gradle.adoc [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+At any point during dev mode, you can attach a Java debugger to debug your application.  By default, the debug port is `7777`, but another port can be used and shown in the console output if the default port is not available.
+
+In your SystemLivenessCheck Class, set a breakpoint inside the call() method.
+
+Attach your IDE's debugger to port 7777.
+For example, in VS Code, click Debug > Add Configuration... > Java: Attach > set "port": "7777". Then View > Debug > select Debug (Attach), then press the arrow icon to start debugging.
+
+In your browser, go to http://localhost:9080/health.
+
+Notice your IDE pauses at the breakpoint that you set, allowing you to debug.
```
Comment (ericglau) [inline drafts/liberty-dev-mode-gradle.adoc]: Instead of providing instructions here, also rephrase this as an example scenario and provide a screenshot if appropriate.  And leave out the VS Code specific steps but just make it more general e.g. "In the example below, we attach our IDE's debugger to port 7777"

File: drafts/liberty-dev-mode-gradle.adoc [hunk 9]
```diff
@@ -0,0 +1,118 @@
+---
+layout: post
+title: "Rapid application development with Gradle and Open Liberty's dev mode"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/u/28316667
+author_github: https://github.com/rzgry
+seo-title: Rapid application development with Gradle and Open Liberty's dev mode - OpenLiberty.io
+seo-description: Develop Java applications quicker using Open Liberty development mode with Gradle.
```
Comment (yeekangc) [inline drafts/liberty-dev-mode-gradle.adoc]:
```suggestion
seo-description: Develop cloud-native Java applications and microservices fast using Open Liberty development mode with Gradle.
```

File: drafts/liberty-dev-mode-gradle.adoc [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+blog_description: "Develop Java applications quicker using Open Liberty development mode with Gradle."
```
Comment (yeekangc) [inline drafts/liberty-dev-mode-gradle.adoc]:
```suggestion
blog_description: "Develop cloudn-native Java applications and microservices fast using Open Liberty development mode with Gradle."
```

File: drafts/liberty-dev-mode-gradle.adoc [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+---
+= Open Liberty development mode for Gradle
+Ryan Zegray <https://github.com/rzgry>
+
+Open Liberty development mode, or dev mode, now works with Gradle projects. Open Liberty dev mode allows you to develop applications with any text editor or IDE by providing hot reload and deployment, on demand testing, and debugger support. Your code is automatically compiled and deployed to your running server, making it easy to iterate on your changes. You can run tests on demand or even automatically so that you can get immediate feedback on your changes. You can also attach a debugger at any time to debug your running application.
```
Comment (yeekangc) [inline drafts/liberty-dev-mode-gradle.adoc]:
```suggestion
Open Liberty development mode, or dev mode, now works with Gradle projects. Open Liberty dev mode allows you to develop cloud-native Java applications with any text editor or IDE by providing hot reload and deployment, on demand testing, and debugger support. Your code is automatically compiled and deployed to your running server, making it easy to iterate on your changes. You can run tests on demand or even automatically so that you can get immediate feedback on your changes. You can also attach a debugger at any time to debug your running application.
```

**Comment by yeekangc** [comment]
@lauracowen, let us see if we can get this out there this week?  =)
